src/testing/seastar_test.cc

```diff
@@ -66,7 +66,7 @@ const std::vector<seastar_test*>& known_tests() {
     return *tests;
 }
 
-seastar_test::seastar_test() {
+seastar_test::seastar_test() noexcept {
```

Comment:
Please add a comment since a reader will see an obvious bug here (noexcept on potentially throwing body).

Also, on what schedule does clang-tidy pay us to fix its reports? I didn't see the check for the previous fixes yet.

Reply:
> Please add a comment since a reader will see an obvious bug here (noexcept on potentially throwing body).

I think one can make a reasonable argument that a possibly-throwing function marked `noexcept` is not a bug: you are just signing up to have your process terminated in case the function would violate this contract. In some cases (and this case) that might be an appropriate action.

That said, I agree that this is kind of distasteful, so here's an alternative which just uses a narrowly targeted `NOLINT`:

https://github.com/scylladb/seastar/pull/1279

> Also, on what schedule does clang-tidy pay us to fix its reports? I didn't see the check for the previous fixes yet.

Did you invoice them promptly?

Reply:
> > Please add a comment since a reader will see an obvious bug here (noexcept on potentially throwing body).
> 
> I think one can make a reasonable argument that a possibly-throwing function marked `noexcept` is not a bug: you are just signing up to have your process terminated in case the function would violate this contract. In some cases (and this case) that might be an appropriate action.

I'm not saying it's a bug, just that a reader will see it as a bug. In fact Seastar uses noexcept blocks in allocating functions.

The comment is important to distinguish between "the noexcept or the allocation are wrong (or maybe both)" and "this noexcept on an allocating block is intentional, and here's why".

> 
> That said, I agree that this is kind of distasteful, so here's an alternative which just uses a narrowly targeted `NOLINT`:
> 
> #1279
> 
> > Also, on what schedule does clang-tidy pay us to fix its reports? I didn't see the check for the previous fixes yet.
> 
> Did you invoice them promptly?

Ah, that's why there's nothing in the mail.